src/main/java/com/jcabi/http/wire/TrustedWire.java

```diff
@@ -109,22 +107,18 @@ public Response send(final Request req, final String home,
         final String method,
         final Collection<Map.Entry<String, String>> headers,
         final InputStream content,
-        final int connect, final int read) throws IOException {
-        synchronized (TrustedWire.class) {
-            final SSLSocketFactory def =
-                HttpsURLConnection.getDefaultSSLSocketFactory();
-            try {
-                HttpsURLConnection.setDefaultSSLSocketFactory(
-                    TrustedWire.context().getSocketFactory()
-                );
-                return this.origin.send(
-                    req, home, method, headers, content,
-                    connect, read
-                );
-            } finally {
-                HttpsURLConnection.setDefaultSSLSocketFactory(def);
-            }
+        final int connect, final int read,
+        final SSLContext context) throws IOException {
+        final SSLContext resolvedContext;
+        if (context == null) {
```

Comment:
@joris-lambrechts If we will start with "default" `SSLContext`, this null-check would probably go away

Reply:
On the other hand, it seems to me now that in `TrustedWire` (which should disable SSL verification) we should disregard incoming `context` completely and replace it with one which comes from `context()` method.

Reply:
@andreoss 
> On the other hand, it seems to me now that in TrustedWire (which should disable SSL verification) we should disregard incoming context completely and replace it with one which comes from context() method.

Then you would not be able to provide your custom context? 

> Also it appears that the whole feature of injection SSLContext could be done with Wire instead of adding a method to Request.

That indeed looks a good option, perhaps you can give it a shot?

Reply:
I removed the if-check in TrustedWire, you made me realize it didn't belong there. 

I also forgot that by adding `com.jcabi.http.Request#sslcontext` it is possible to define the SSLContext. I think you then should use the TrustedWire anymore. The use of TrustedWire ignores SSL PKIX verifications. If you want a SSLContext for a different use-case, that can now be done without the need of a new Wire implementation.

I added a test case for both the JDKRequest and ApacheRequest to show how it works.